### ISSUE TITLE
[Bug]: Tiny x speed after entering left/right-facing portals with 0 x speed and positive y speed, interfering with retained speed refund

### ISSUE THREAD

**Issue by Haku-Runedisk** [CommunalHelper/OutbackHelper#6]
### Before You Report

- [X] My mods and Everest install are up to date.
- [ ] I have recreated the bug with a minimum number of mods installed.

### What platform are you playing on?

Linux

### Bug Description

Entered a left-facing portal on the same frame as wall collision, resulting in retained speed and 0 x speed. Transported to right-facing portal, so retained speed is expected to be refunded.
Result:
If having y speed <= 0 (so not downwards), tiny positive (or 0) x speed is given, which doesn't interfere with refunding retained speed, and speed is refunded as expected.
If having y speed > 0 (moving downwards), tiny negative x speed is given, cancelling retained speed and removing your speed.
Example values, (x, y) -> (x, y):
(0, 7.5) -> (-0.00000131, 7.5)
(0, 0) -> (0.00000000, 0)
(0, -60) -> (0.00001049, -60)

In other words:
Expectation for behavior: to have x speed after being transported be exactly 0, if x speed entering the portal is 0.

### Steps to Reproduce

From TAS puzzle map Speed Trials' room "secret":
https://github.com/user-attachments/assets/6e167571-1649-4af7-aef3-06544f99fc28
